plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1296,13 +1295,12 @@ private static Location getValidatedExternalLocation(String location)
         return validated;
     }
 
-    private void checkExternalPath(HdfsContext context, Path path)
+    private void checkExternalPathAndCreateIfNotExists(HdfsContext context, Path path)
     {
         try {
-            if (!isS3FileSystem(context, hdfsEnvironment, path)) {
-                if (!hdfsEnvironment.getFileSystem(context, path).isDirectory(path)) {
-                    throw new TrinoException(INVALID_TABLE_PROPERTY, "External location must be a directory: " + path);
-                }
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, path);
+            if (!fileSystem.isDirectory(path)) {
+                HiveWriteUtils.createDirectory(context, hdfsEnvironment, path);
```

Comment:
Do we still throw this error `INVALID_TABLE_PROPERTY, "External location must be a directory: " + path`  - if it points to a file ?

Reply:
No, HiveWriteUtils will throw 
`throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed to create directory: " + path, e);`
if we point to file